### PR TITLE
test: raise backend statement coverage from 75.6% to 81.1%

### PR DESCRIPTION
**Pull request by goshacodes**
## Summary

Adds ZIO-test coverage for previously untested repositories, services, controllers and orchestration branches, identified via scoverage gap analysis (per-file uncovered-statement counts across all 8 modules).

**Aggregate: 75.63% → 81.06% statements (12045 → 12564 of 15500). All 8 modules green, 0 test failures.**

| module | before | after |
|---|---|---|
| auth | 76.4% | 80.6% |
| auth-postgres | 77.7% | 84.0% |
| central | 73.9% | 79.7% |
| central-postgres | 69.0% | 80.3% |
| edge | 74.6% | 83.3% |
| edge-postgres | 34.8% | 91.3% |
| util | 78.9% | 81.7% |
| util-postgres | 84.4% | 84.4% |

## What's covered now

- **Postgres repositories with zero prior tests**: `EdgeSessionRepository`, `LoginRepository`, `JwksRepository`, `ChallengeSettingsRepository`, `ThemeRepository` — following the existing shared-trait + `PostgresSpec` pattern used elsewhere in the repo.
- **Controllers with real branching that were untested**: `MetadataController`, `ServerMetadataController`, `AuthorizationDetailTypeController`, `ServiceController` (both `auth` and `central`).
- **`ConversationService.setNewPassword` / `offerSetPassword`**: the temporary-password flow's ban/rate-limit/reuse/write-conflict branches.
- **`AuthorizeRequestParser` login_hint rejection paths**: disallowed credential type, disallowed phone prefix, unrestricted prefixes, malformed email/phone.
- **`PostgresUserRepository` gaps** in both `auth` (register, findByLogin, patchClaims, delete, findRolesByUser) and `central` (patch, enqueueRoleUpdate, delete, and the outbox lifecycle — including a real behavioral detail: the outbox is per-user FIFO, so a leased event blocks every later event for that user; tests now drain via claim+ack).
- Assorted edge-case coverage: `AuthClient`, `SSOClient`, `PermissionService`, `JWT` claim conversion (all Java type branches), `CleanupManager` (interruption propagation on `stop()`), session cookies.

## What's left for 85%

The remaining gap is concentrated almost entirely in `ConversationService.scala` (487 of ~611 missing statements) — a long tail of rate-limit × ban × write-conflict permutations across the passkey enrollment, registration, consent and code-issuance flows. Each test is cheap individually but there are on the order of 80–120 of them, and several are close to asserting that a stub returns what it was told to return. Happy to continue in a follow-up if 85% is a hard requirement; otherwise this is a reasonable place to gate coverage (e.g. `coverageMinimumStmtTotal := 81`) and ratchet up as those flows get touched.

## Testing

Ran the full suite sequentially against a local Postgres 15 (`sbt coverage test coverageReport coverageAggregate`, `POSTGRES_HOST=localhost:5432`) — all modules pass with 0 failures. Note: `PostgresNotificationListenerSpec` is flaky under parallel execution on a single-core box; sequential execution was used as a workaround. CI's runner (multi-core, dedicated Postgres service container) may not hit this, but flagging it in case it does.

---
Pull Request opened by [Augment Code](https://www.augmentcode.com/) | [View session](https://cosmos.augmentcode.com/session?agentId=01M1PC5CH88GK06TD3XTRQSRFA&panel=chat)

### CODE REVIEW

File: auth/src/test/scala/versola/oauth/model/CookiesSpec.scala [hunk 1]
```diff
@@ -44,51 +51,109 @@ object CookiesSpec extends ZIOSpecDefault:
...
+        assertTrue(UserAgentCookie.parse(tampered, secret).isLeft)
+      },
+      test("parse rejects content with no signature separator") {
+        assertTrue(UserAgentCookie.parse("no-signature-here", secret) == Left("missing signature"))
+      },
+      test("parse rejects a signed payload that is not the expected JSON shape") {
+        // Signed with the right key, so this gets past the MAC check and fails on decoding.
+        val bytes = """{"unexpected":true}""".getBytes("UTF-8").nn
+        val signed = UserAgentCookie(userAgentId, userAgentData, 180.days, secret).content
+        val tampered = s"${Base64.urlEncode(bytes)}.${signed.split('.')(1)}"
```
[the remark this answers is not in the document]
Comment (goshacodes) [reply auth/src/test/scala/versola/oauth/model/CookiesSpec.scala]: Correct, and the bare `isLeft` meant it passed either way. Fixed in 1d39e40.

The blocker was that `computeMac` is private, so a test can't re-sign an arbitrary payload from outside. But all three cookie objects derive their signature from the same keyed hash over the payload bytes with the same `Secret.Bytes32`, so a `ConversationCookie`'s own content already carries a signature `UserAgentCookie.parse` accepts — while its payload is not a `UserAgentCookiePayload`. That gets past the MAC check and lands on the decoding step.

Also replaced the bare `isLeft` so it can't silently stop short again:

```scala
val result = UserAgentCookie.parse(content, secret)
assertTrue(
  result.isLeft,
  result != Left("invalid signature"),
  result != Left("missing signature"),
)
```

Those two assertions are what make it non-vacuous — they'd have failed on the old construction, which produced exactly `Left("invalid signature")`. Kept off the exact zio-json message to avoid over-fitting to its wording. `CookiesSpec` is 16/16 green.

I checked the rest of the PR for the same class of problem: the other `.isLeft` assertions all target failure modes where the first failure reached *is* the intended one (wrong secret → MAC mismatch, non-base64 payload → base64 decode), so this was the only test asserting a later branch than it actually got to.

One note on the coverage number in the PR body: 81.06% was measured before this change. This can only have nudged it up, since the decode statement in `UserAgentCookie.parse` now gets hit where it previously didn't, but I haven't re-run the full aggregate to confirm — the sandbox I'm in got re-provisioned without a JDK or Postgres, so I installed the toolchain to run `CookiesSpec` but the DB-backed specs would need a Postgres 15 as well.
